modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -159,8 +159,8 @@ void writeBackExec(const Mag& mag, const RcDesc &rc, GRunArgP &g_arg)
         // a real copy (add a pass to StreamingBackend?)
         auto& out_mat = *util::get<cv::Mat*>(g_arg);
         const auto& rmat = mag.template slot<cv::RMat>().at(rc.id);
-        auto mag_data = rmat.get<RMatAdapter>()->data();
-        if (out_mat.data != mag_data) {
+        auto* adapter = rmat.get<RMatAdapter>();
+        if (adapter != nullptr && out_mat.data != adapter->data()) {
```

Comment:
> `adapter != nullptr`

Should this to be an assertion with raised exception in case of null?

Reply:
No
`adapter == nullptr` is a normal case.
We need to do copy only if adapter is RMatAdapter and `out_mat.data != adapter->data()`